cmd/cvp-trigger/main.go

```diff
@@ -4,69 +4,190 @@ import (
 	"flag"
 	"fmt"
 	"os"
+	"sort"
+	"strings"
 
 	"github.com/ghodss/yaml"
+	"github.com/openshift/ci-tools/pkg/util"
 	"github.com/sirupsen/logrus"
+	"github.com/spf13/afero"
+	v1 "k8s.io/api/core/v1"
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 	"k8s.io/apimachinery/pkg/fields"
-	"k8s.io/client-go/rest"
-
 	pjapi "k8s.io/test-infra/prow/apis/prowjobs/v1"
 	pjclientset "k8s.io/test-infra/prow/client/clientset/versioned"
 	prowconfig "k8s.io/test-infra/prow/config"
 	"k8s.io/test-infra/prow/interrupts"
 	"k8s.io/test-infra/prow/pjutil"
-
-	"github.com/openshift/ci-tools/pkg/util"
 )
 
 const (
-	prowConfigPathOption = "prow-config-path"
-	jobConfigPathOption  = "job-config-path"
-	periodicOption       = "periodic"
+	bundleImageRefOption   = "bundle-image-ref"
+	channelOption          = "channel"
+	indexImageRefOption    = "index-image-ref"
+	installNamespaceOption = "install-namespace"
+	jobConfigPathOption    = "job-config-path"
+	jobNameOption          = "job-name"
+	ocpVersionOption       = "ocp-version"
+	outputFilePathOption   = "output-path"
+	packageNameOption      = "package-name"
+	prowConfigPathOption   = "prow-config-path"
+	releaseImageRefOption  = "release-image-ref"
+	targetNamespacesOption = "target-namespaces"
 )
 
 type options struct {
-	confirm bool
+	bundleImageRef   string
+	channel          string
+	indexImageRef    string
+	installNamespace string
+	jobConfigPath    string
+	jobName          string
+	ocpVersion       string
+	outputPath       string
+	packageName      string
+	prowConfigPath   string
+	releaseImageRef  string
+	targetNamespaces string
+	dryRun           bool
+}
 
-	prowConfigPath string
-	jobConfigPath  string
+var fileSystem = afero.NewOsFs()
```

Comment:
`fileSystem` is available for substitution in testing. In our test we set `filesystem = afero.NewMemMapFs()` so that we're able to create arbitrary paths for path existence validations.